From 819cf0718cce20a61768fcbf11cc737fd29f407b Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 20 Sep 2012 13:16:16 +0800
Subject: [PATCH 2/8] ENGR00224997-2 vold: refine the return value check of
 fat fsck

When the volume checking each partitions for avaliable
VFAT filesystem, we will get ENOEXEC return value on
exec msdos_fsck who give out "Invalid cluster size: 0"
on the extension partition.
This error return, should not block volume manager
to scan the left partitions as it's not a I/O error.
This issue happend when we have a extension partition
on external SD card.

Change-Id: I5f9e269c6d07fa58ef19a4be2e216ece1a9d12e0
Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 Fat.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Fat.cpp b/Fat.cpp
index c967a90..33342d8 100644
--- a/Fat.cpp
+++ b/Fat.cpp
@@ -101,6 +101,10 @@ int Fat::check(const char *fsPath) {
             SLOGE("Failing check after too many rechecks");
             errno = EIO;
             return -1;
+        case 8:
+            SLOGW("We open the extension partition");
+            errno = ENODATA;
+            return -1;
 
         default:
             SLOGE("Filesystem check failed (unknown exit code %d)", status);
-- 
1.8.0

